File: Distinct.py
# you can write to stdout for debugging purposes, e.g.
# print "this is a debug message"

def solution(A):
    # write your code in Python 2.7
    if len(A)==0:
        return 0;
    dict = {}
    res = A[0]
    for i in A:
        if dict.has_key(i):    
            dict[i]+=1
        else:
            dict[i]=1
    return len(dict)

File: Distinct.cpp
// you can use includes, for example:
// #include <algorithm>
#include <map>

// you can write to stdout for debugging purposes, e.g.
// cout << "this is a debug message" << endl;

int solution(vector<int> &A) {
    // write your code in C++11 (g++ 4.8.2)
    int count = 0;
    if (A.size() == 0)
        return count;
        
    map<int,int> dic;
    for(auto &i: A)
    {
        ++dic[i];
    }
    count = dic.size();
    return count;
}